.github/workflows/linux_unit_tests_with_latest_deps.yml

```diff
@@ -9,24 +9,76 @@ on:
 
 jobs:
   unit_tests:
-    name: ${{ matrix.python_version }} unit tests, Core Dependencies (${{ matrix.core_dependencies }}), Codecov (${{ matrix.codecov }})
+    name: ${{ matrix.python_version }} unit ${{matrix.dir}} tests, Core Dependencies (${{ matrix.core_dependencies }}), Codecov (${{ matrix.codecov }})
     runs-on: ubuntu-latest
     strategy:
-      fail-fast: true
+      fail-fast: false
```

Comment:
Do you just want to set `fail-fast` to `true` for all jobs? The thing I don't like about moving tests outside the test matrix is that it leads to a lot of duplicated code across the yamls because the install steps are the same. There may be a way to reuse code across actions but I'd rather just `fail-fast` to alleviate concerns over too many jobs running.